Tools/Create-Azure-Sentinel-Solution/common/commonFunctions.ps1

```diff
@@ -0,0 +1,3285 @@
+                                throw "The file $fileName has metadata with source -> kind = Community | Standalone. Please remove it so that it can be packaged as a solution."
+                            }
+						}
+						$workbookUIParameter = [PSCustomObject] @{ name = "workbook$global:workbookCounter"; type = "Microsoft.Common.Section"; label = $dependencies.title; elements = @( [PSCustomObject] @{ name = "workbook$global:workbookCounter-text"; type = "Microsoft.Common.TextBlock"; options = @{ text = $dependencies.description; } } ) }
+                        $global:baseCreateUiDefinition.parameters.steps[$global:baseCreateUiDefinition.parameters.steps.Count - 1].elements += $workbookUIParameter
+
+                        try {
+                            $data = $rawData
+                            # Serialize workbook data
+                            $serializedData = $data |  ConvertFrom-Json -Depth $jsonConversionDepth
+                            # Remove empty braces
+                            $serializedData = $(removePropertiesRecursively $serializedData) | ConvertTo-Json -Compress -Depth $jsonConversionDepth | Out-String
+                        }
+                        catch {
+                            Write-Host "Failed to serialize $file" -ForegroundColor Red
+                            break;
+                        }
+                        $workbookDescriptionText = $(if ($contentToImport.WorkbookDescription -and $contentToImport.WorkbookDescription -is [System.Array]) { $contentToImport.WorkbookDescription[$global:workbookCounter - 1] } elseif ($contentToImport.WorkbookDescription -and $contentToImport.WorkbookDescription -is [System.String]) { $contentToImport.WorkbookDescription } else { "" })
+                        #creating parameters in mainTemplate
+                        $workbookIDParameterName = "workbook$global:workbookCounter-id"
+                        $workbookNameParameterName = "workbook$global:workbookCounter-name"
+
+                        $workbookIDParameter = [PSCustomObject] @{ type = "string"; defaultValue = "[newGuid()]"; minLength = 1; metadata = [PSCustomObject] @{ description = "Unique id for the workbook" }; }
+
+                        if(!$contentToImport.TemplateSpec)
+                        {
+                            $global:baseMainTemplate.parameters | Add-Member -MemberType NoteProperty -Name $workbookIDParameterName -Value $workbookIDParameter
+                        }
+
+                        # Create Workbook Resource Object
+                        $newWorkbook = [PSCustomObject]@{
+                            type       = "Microsoft.Insights/workbooks";
+                            name       = "[parameters('workbook$global:workbookCounter-id')]";
+                            location   = "[parameters('workspace-location')]";
+                            kind       = "shared";
+                            apiVersion = "2021-08-01";
```

Comment:
Can we move these hard coded api version values to the dict object so that in future it would be easy to change at 1 place

Reply:
Created a seperate PBI in ADO to handle this https://dev.azure.com/SentinelThirdPartyEcosystem/Sentinel%20TCS/_workitems/edit/5895